lib/logstash/filters/dictionary/json_file.rb

```diff
@@ -0,0 +1,84 @@
+# encoding: utf-8
+require "json"
+
+module LogStash module Filters module Dictionary
+  class JsonFile < File
+    def initialize_for_file_type
+    end
+
+    def read_file_into_dictionary
+      content = IO.read(@dictionary_path, :mode => 'r:bom|utf-8')
+      @dictionary.update(::JSON.load(content)) unless content.nil? || content.empty?
+    end
+  end
+end end end
+
+__END__
+Preserving the text below for near term prosperity...
+
+I tried hard to find a stream parsing solution with JrJackson and sc_load
+but it was no faster than the above code.
+The idea is for each line to be read into the streaming parse that will update
+the @dictionary as each key/value is found.
+It will be lower on memory consumption because the JSON string is not read into memory
+and then a Ruby Hash created and merged into @dictionary.
+I decided to trade speed for memory. Side Note, it seems that
+the json gem has become quite speedy lately.
+
+e.g.
+require_relative 'json_handler'
+...
+    def initialize_for_file_type
+      @handler = JsonHandler.new(@dictionary)
+    end
+
+    def read_file_into_dictionary
+      ::File.open(@dictionary_path, "r:bom|utf-8") do |io|
+        JrJackson::Json.sc_load(@handler, io, {raw: true})
+      end
+    end
+...
+where JsonHandler is:
+
+require 'jrjackson'
+
+module LogStash module Filters module Dictionary
+  class JsonHandler
```

Comment:
All of this section is below a `__END__` point.

It is recorded here in case I want to use it later.

I wanted to give `json_file` the same "streaming" style of dictionary load as csv and yaml but there were no performance gains in using JrJackson 😢 and so `I have a long term goal of removing deps on JrJackson in LS` applied.

I looked at whether the JSON gem had recently acquired "streaming" support but it has not.

An alternative is to use a similar Jackson construct as shown it the StreamingJson experiment PR I put up but that is the subject of another PR.